### ISSUE TITLE
Task graph-to-dsl conversion possibly broken

### ISSUE THREAD

**Issue by jvalkeal** [spring-cloud/spring-cloud-dataflow#1712]
Simple one node `timestamp` gives dsl as `timestamp && END`.
For `/tools/convertTaskGraphToText`, request/reponse is new UI vs OLD

```
{
  "nodes": [
    {
      "id": "c21b4919-268c-4e20-8424-4f8d2daacc74",
      "name": "START",
      "metadata": {
        
      }
    },
    {
      "id": "4d47d684-fd65-4c39-a556-51f56c2dd8c5",
      "name": "timestamp",
      "metadata": {
        
      }
    },
    {
      "id": "e56b1443-0c5b-4246-bc17-5fc7948eb974",
      "name": "END",
      "metadata": {
        
      }
    }
  ],
  "links": [
    {
      "from": "c21b4919-268c-4e20-8424-4f8d2daacc74",
      "to": "4d47d684-fd65-4c39-a556-51f56c2dd8c5",
      "properties": {
        
      }
    },
    {
      "from": "4d47d684-fd65-4c39-a556-51f56c2dd8c5",
      "to": "e56b1443-0c5b-4246-bc17-5fc7948eb974",
      "properties": {
        
      }
    }
  ]
}

{"graph":null,"dsl":"timestamp && END","errors":[]}
```
```
{
  "nodes": [
    {
      "name": "START",
      "id": "718cc3d0-795c-40aa-a7c2-087d95b19031"
    },
    {
      "name": "END",
      "id": "d0e415c0-7977-4f25-9b9c-4ac443bbd280",
      "properties": {
        
      }
    },
    {
      "name": "timestamp",
      "id": "9c4c0590-38f2-428e-a9ae-8f56075962d2",
      "properties": {
        
      }
    }
  ],
  "links": [
    {
      "from": "718cc3d0-795c-40aa-a7c2-087d95b19031",
      "to": "9c4c0590-38f2-428e-a9ae-8f56075962d2"
    },
    {
      "from": "9c4c0590-38f2-428e-a9ae-8f56075962d2",
      "to": "d0e415c0-7977-4f25-9b9c-4ac443bbd280"
    }
  ],
  "properties": {
    
  }
}

{"graph":null,"dsl":"timestamp","errors":[]}
```

**Comment by jvalkeal** [comment]
Here's a simple curl commands to replicated this, old vs. new
```
curl 'http://localhost:9393/tools/convertTaskGraphToText' -H 'Content-Type: application/json;charset=UTF-8' -H 'Accept: application/json, text/plain, */*' --data-binary '{"nodes":[{"name":"START","id":"0a179ae1-ed0e-4e0c-8e79-5322bf79e59f"},{"name":"END","id":"81348a97-68fe-409c-aad1-b5e67bbabda1","properties":{}},{"name":"timestamp","id":"757ba6f9-b474-4d83-bc19-14d4a196ba88","properties":{}}],"links":[{"from":"0a179ae1-ed0e-4e0c-8e79-5322bf79e59f","to":"757ba6f9-b474-4d83-bc19-14d4a196ba88"},{"from":"757ba6f9-b474-4d83-bc19-14d4a196ba88","to":"81348a97-68fe-409c-aad1-b5e67bbabda1"}],"properties":{}}'

{"graph":null,"dsl":"timestamp","errors":[]}

curl 'http://localhost:9393/tools/convertTaskGraphToText' -H 'Content-Type: application/json;charset=UTF-8' -H 'Accept: application/json, text/plain, */*' --data-binary '{"nodes":[{"id":"c21b4919-268c-4e20-8424-4f8d2daacc74","name":"START","metadata":{}},{"id":"4d47d684-fd65-4c39-a556-51f56c2dd8c5","name":"timestamp","metadata":{}},{"id":"e56b1443-0c5b-4246-bc17-5fc7948eb974","name":"END","metadata":{}}],"links":[{"from":"c21b4919-268c-4e20-8424-4f8d2daacc74","to":"4d47d684-fd65-4c39-a556-51f56c2dd8c5","properties":{}},{"from":"4d47d684-fd65-4c39-a556-51f56c2dd8c5","to":"e56b1443-0c5b-4246-bc17-5fc7948eb974","properties":{}}]}'

{"graph":null,"dsl":"timestamp && END","errors":[]}
```

**Comment by jvalkeal** [comment]
So if I remove metadata and properties from json, then it creates correct dsl
```
{
  "nodes": [
    {
      "id": "c21b4919-268c-4e20-8424-4f8d2daacc74",
      "name": "START"
    },
    {
      "id": "e56b1443-0c5b-4246-bc17-5fc7948eb974",
      "name": "END"
    },
    {
      "id": "4d47d684-fd65-4c39-a556-51f56c2dd8c5",
      "name": "timestamp"
    }
  ],
  "links": [
    {
      "from": "4d47d684-fd65-4c39-a556-51f56c2dd8c5",
      "to": "e56b1443-0c5b-4246-bc17-5fc7948eb974"
    },
    {
      "from": "c21b4919-268c-4e20-8424-4f8d2daacc74",
      "to": "4d47d684-fd65-4c39-a556-51f56c2dd8c5"
    }
  ]
}

{"graph":null,"dsl":"timestamp","errors":[]}
```

**Comment by aclement** [comment]
Thanks to Jannes observations I narrowed it down quick.  The code was assuming if properties was set for the link to END that there was a need to include END in the DSL (which is true if there is a transition name in that properties object) but it wasn't differentiating between properties being null and properties being set but empty. I'll admit I haven't built the backend with this change and used curl or the UI to drive it to check the problem is fixed but I wrote a good testcase and am reasonably confident this is what it is.
